cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -145,18 +145,49 @@ instance ElidingTracer (WithSeverity (ChainDB.TraceEvent blk)) where
   isEquivalent (WithSeverity s1 (ChainDB.TraceGCEvent ev1))
                (WithSeverity s2 (ChainDB.TraceGCEvent ev2)) =
                   s1 == s2 && indexGCType ev1 == indexGCType ev2
-   
+  isEquivalent (WithSeverity _s1 (ChainDB.TraceAddBlockEvent _))
+               (WithSeverity _s2 (ChainDB.TraceAddBlockEvent _)) = True
+  isEquivalent (WithSeverity _s1 (ChainDB.TraceGCEvent _ev1))
+               (WithSeverity _s2 (ChainDB.TraceAddBlockEvent _)) = True
```

Comment:
So you're saying that `TraceGCEvent` and `TraceAddBlockEvent` are equivalent so that when a `TraceGCEvent`, which is unrelated to the `TraceAddBlockEvent` in question, doesn't interrupt the elision and is elided as well? I guess that works, but it feels strange/ad hoc/hacky to me.

What about the following: instead of eliding *consecutive* "equivalent" messages (i.e., not interrupted by "un-equivalent" messages), just print those "un-equivalent" messages without interrupting the elision by making the elision a bit cleverer. Instead of looking at the equivalence with the previous message, remember whether an equivalent message was traced in the last x milliseconds or so. You'll have to be careful not to hold on to too much memory. This means that the elision doesn't restart when interrupted and the unrelated "un-equivalent" (that are incorrectly made "un-equivalent" in this PR) still show up in the output.

Reply:
The logic of the message suppression is that simple now. I hesitate to make it much more complex, e.g. keep more state, and potentially run into problems. We can't compare timestamps as they are not part of the state. Here we only compare based on the traced types and the annotated severity.

Reply:
I think that if we want to use such broad equivalence for elision, then we probably want to make elision optional -- for the cases when we need the full message stream -- for debugging purposes.

What do you think @CodiePP ?

Reply:
the 'doelide' functions can be commented out to disable eliding.